chrome/content/zotero/xpcom/plugins.js

```diff
@@ -57,22 +60,63 @@ Zotero.Plugins = new function () {
 			registerLocales(addon);
 			await _callMethod(addon, 'startup', REASONS.APP_STARTUP);
 		}
-		
+
 		Zotero.addShutdownListener(async () => {
 			var { addons } = await AddonManager.getActiveAddons(["extension"]);
 			for (let addon of addons) {
 				await _callMethod(addon, 'shutdown', REASONS.APP_SHUTDOWN);
 			}
 		});
+
+		const mainWindowListener = {
+			onOpenWindow: function (aWindow) {
+				const domWindow = aWindow
+					.QueryInterface(Ci.nsIInterfaceRequestor)
+					.getInterface(Ci.nsIDOMWindowInternal || Ci.nsIDOMWindow);
+				async function onload() {
+					domWindow.removeEventListener("load", onload, false);
+					if (
+						domWindow.location.href
+						!== "chrome://zotero/content/zoteroPane.xhtml"
+					) {
+						return;
+					}
+					const { addons } = await AddonManager.getActiveAddons(["extension"]);
+					for (const addon of addons) {
+						await _callMethod(addon, 'onMainWindowLoad', REASONS.MAIN_WINDOW_LOAD, { window: domWindow });
+					}
+				}
+				domWindow.addEventListener(
+					"load",
+					onload,
+					false,
+				);
```

Comment:
Can just be on one line unless it's >100 chars (and same throughout)

Reply:
Done